amaranth/lib/io.py

```diff
@@ -476,6 +620,16 @@ def elaborate(self, platform):
             else:
                 m.submodules += IOBufferInstance(self._port.p, o=o_inv, oe=self.oe, i=i_inv)
                 m.submodules += IOBufferInstance(self._port.n, o=~o_inv, oe=self.oe)
+        elif isinstance(self._port, SimulationPort):
+            if self.direction in (Direction.Input, Direction.Bidir):
+                # Loop back `o` if `oe` is asserted. This frees the test harness from having to
+                # provide this functionality itself.
+                for i_inv_bit, oe_bit, o_bit, i_bit in \
+                        zip(i_inv, self._port.oe, self._port.o, self._port.i):
```

Comment:
this would cause an `AttributeError` on ports with `Input` direction, wouldn't it?

Reply:
Oh, hm. I wonder why are the tests not catching that?

Reply:
Fixed